jina/docker/hubapi.py

```diff
@@ -1,112 +1,104 @@
-
 import json
-import requests
-from typing import Dict
+from typing import Dict, Sequence, Any, Optional
+from urllib.error import HTTPError
+from urllib.parse import urlencode
+from urllib.request import Request, urlopen
+
 from pkg_resources import resource_stream
 
 from .helper import credentials_file
 from ..helper import yaml, colored
+from ..logging.profile import TimeContext
 
 
-def _list(logger, name: str = None, kind: str = None, type_: str = None, keywords: tuple = ('numeric')):
-    """ Hub API Invocation to run `hub list` """
+def _list(logger, image_name: str = None, image_kind: str = None,
```

Comment:
I wondered, why we don't use `**kwargs` here as we do all around the code. I guess, since it is not obvious, what the kwargs should be.

Anyhow, if this method would be a static method in the `HubIO` class using `**kwargs` would make way more sense. And since it is only used in `HubIO` it is also well placed there.

Reply:
These arguments are highly specific to the hubapi, hence passing explicitly makes more sense. This method is kept out of `HubIO`, as we can keep all hubapi based invocations in one place. This file will evolve as we add more endpoints.

Reply:
But then it should not be "private" and we should remove the underscore in front.